core/matchers/matcher.ts

```diff
@@ -50,6 +50,11 @@ export class Matcher<T> {
 
     if (expectedValue instanceof TypeMatcher) {
       valueMatch = expectedValue.test(this._actualValue);
+    } else if (
+      Buffer.isBuffer(expectedValue) ||
+      Buffer.isBuffer(this._actualValue)
```

Comment:
If `Buffer.isBuffer` is `true` for one of the values and `false` for the other, we can safely assume that they don't match and return false.

Reply:
I had originally implemented it that simply, but after researching some, I had thought to allow comparing to types that are bufferable.

Check out my test cases to see what I mean.

Reply:
I think it should only be necessary to test for the expected value in this case we shouldn't need to check whether the actual value is as I'd expect that

```typescript
Expect(1).toEqual(Buffer.from([1]));
```

To throw based on the `==` comparison which it should so that should be fine :)

Other than that great PR 👍 just that small tweak and let's get it out there!

Reply:
Makes sense. I'll do this change tonight.